src/payment/order-summary/Offers.jsx

```diff
@@ -16,23 +16,30 @@ Benefit.propTypes = {
   benefitValue: PropTypes.number.isRequired,
 };
 
+function Message({ benefitType, benefitValue, provider }) {
+  if (provider) {
+    return [<Benefit benefitType={benefitType} benefitValue={benefitValue} />, ' discount provided by', provider];
```

Comment:
If I'm reading this code right, 'discount provided by' and 'discount for your first upgrade applied' aren't localized anymore...  Might be more clearly factored by taking the conditional in `Message` and just inserting it into `Offer` directly and choosing one of two `<FormattedMessage>` components to insert into the paragraph tag.

Something like (I have not tested this code):

```
function Offer({
  benefitType, benefitValue, provider,	  benefitType, benefitValue, provider,
}) {
  let message = null;
  if (provider) {
    message = <FormattedMessage
      id="payment.summary.discount.offer"
      defaultMessage="{benefit} discount provided by {provider}."
      description="A description of a discount offer applied to a basket."
      values={{
          benefit: (
            <Benefit
              benefitType={benefitType}
              benefitValue={benefitValue}
            />
          ),
          provider,	
        }}
      />;
    } else {
      message = <FormattedMessage
        id="payment.summary.discount.offer"
        defaultMessage="{benefit} discount for your first upgrade applied."
        description="A description of a discount offer applied to a basket for your first upgrade."
        values={{
          benefit: (
            <Benefit
              benefitType={benefitType}
              benefitValue={benefitValue}
            />
          ),
        }}
      />;
    }
  return (
    <p className="m-0 text-muted" key={`${benefitValue}-${provider}`}>	    <p className="m-0 text-muted" key={`${benefitValue}-${provider}`}>
      {message}
    </p>
  );
}
```

Reply:
(obviously the <FormattedMessage> itself could be DRY'd up a bit, but this way we can just do it using the existing components)

Reply:
I'm not familiar with how formatted message works, this is the method that does the internationalization here?
Is the issue that because the text being inserted is a variable/react component, it won't be internationalized?  
If so, could I just call FormattedMessage directly on the strings that are currently not internationalized in their current location?

Reply:
Usages of `<FormattedMessage />` are grepped for by a script at build time, and the defaultMessage, description, and ID are bundled off to Transifex for translation along with files of the pattern `*.messages.js`.  So any text that's not part of one - i.e, the text above in `Message`, won't be internationalizable.

The rest of my suggestion is just a more canonical/standard way of factoring the code in light of the above need to make things internationalizable, IMHO.

Reply:
makes sense
i've tried to implement your suggestion above
it seems like this would be more total lines than if the message was factored out into two formatted messages in its own function, but if this is the more canonical way to do it then that's fine